### PR TITLE
tlogbe: hook up token prefixes in comments & ticketvote plugins

### PR DESCRIPTION
**Pull request by amass01**

### CODE REVIEW

File: politeiad/backend/tlogbe/comments.go [hunk 1]
```diff
@@ -121,7 +121,11 @@ func (p *commentsPlugin) mutex(token string) *sync.Mutex {
 	return m
 }
 
+// commentsIndexPath accepts full length token or token prefix but always
+// uses prefix when generating the comments index path string.
 func (p *commentsPlugin) commentsIndexPath(s comments.StateT, token string) string {
+	// Use token prefix
+	token = tokenPrefix([]byte(token))
```
Comment (lukebp) [inline politeiad/backend/tlogbe/comments.go]: You can't decode hex like this.

File: politeiad/backend/tlogbe/tlogclient.go [hunk 2]
```diff
@@ -129,6 +144,18 @@ func (c *backendClient) blobsByMerkle(tlogID string, token []byte, merkles [][]b
 		return nil, err
 	}
 
+	if len(token) == tokenPrefixSize() {
```
Comment (lukebp) [inline politeiad/backend/tlogbe/tlogclient.go]: Why do you need this here is you've already implemented it in `treeIDFromToken()`?

**Review by lukebp — CHANGES_REQUESTED** [review]

File: politeiad/backend/tlogbe/comments.go (hunk 1 above)
Comment (amass01) [reply politeiad/backend/tlogbe/comments.go]: my bad

File: politeiad/backend/tlogbe/tlogclient.go (hunk 2 above)
Comment (amass01) [reply politeiad/backend/tlogbe/tlogclient.go]: no need

**Comment by amass01** [comment]
@lukebp 

I just saw you merged https://github.com/lukebp/politeia/pull/50, some comments tests may fail without this one as they are using token prefixes.

File: politeiad/backend/tlogbe/ticketvote.go [hunk 3]
```diff
@@ -405,16 +405,27 @@ func (p *ticketVotePlugin) cachedVotesDel(token string) {
 	log.Debugf("ticketvote: deleted votes cache: %v", token)
 }
 
-func (p *ticketVotePlugin) cachedSummaryPath(token string) string {
+// cachedSummaryPath accepts both full tokens and token prefixes, however it
+// uses always the token prefix when generatig the path.
```
Comment (lukebp) [inline politeiad/backend/tlogbe/ticketvote.go]: typo

```suggestion
// always uses the token prefix when generatig the path.
```

**Review by lukebp — CHANGES_REQUESTED** [review]

**Comment by lukebp** [comment]
@amassarwi thanks.  This looks good too.  Just the one typo.

**Comment by amass01** [comment]
@lukebp updated

**Review by lukebp — APPROVED** [review]

**Pull request merged**